crates/core/src/operations/transaction/mod.rs

```diff
@@ -604,14 +604,22 @@ impl<'a> std::future::IntoFuture for PreparedCommit<'a> {
             // unwrap() is safe here due to the above check
             let read_snapshot = this.table_data.unwrap().eager_snapshot();
 
-            let mut attempt_number = 1;
-            while attempt_number <= this.max_retries {
+            let mut attempt_number = 0;
```

Comment:
```suggestion
            let mut attempt_number = 1;
```
Should start at one btw

Reply:
@daphnenhuch-at you resolved this, but this was still required so I committed it manually.

Since we do attempt_number <= max.retries + 1, we should counting at 1, since max_retries would be attempt number 2 that will still pass (1 retry) and attempt number 3 (retry 2) it will fail.